app/src/main/java/com/globant/practice/presentation/view/fragment/WebClientFragment.java

```diff
@@ -130,17 +130,10 @@ public void onResume() {
     @Override
     public void render(@NonNull WebClientState webClientState) {
         if (webViewState != null) {
-            if (webClientState.isLoading()) {
-                webPageLoadingIndicator.setMessage(String.format(getString(R.string.web_client_progress_dialog_message), webClientState.getDetailType()));
-                webPageLoadingIndicator.show();
-                webClientWebView.restoreState(webViewState);
-                webViewState = null;
-            } else {
-                webClientWebView.restoreState(webViewState);
-                webViewState = null;
-            }
+            webClientWebView.restoreState(webViewState);
+            webViewState = null;
+            webPageLoadingIndicator.show();
```

Comment:
Yes, when the restoreState is call the WebView has the same behavior like when call loadUrl with onPageFinished or onReceivedError calls.